modules/drivers/j2k/openjpeg/wscript

```diff
@@ -16,128 +16,129 @@ def options(opt):
 def configure(conf):
     j2kLayer = Options.options.j2k_layer
     j2kHome = Options.options.j2k_home
-    if j2kLayer == 'openjpeg' and not j2kHome:
 
-        # callback function to check for #defines used by the openjpg module
-        def opj_callback(conf):
-
-            # check functionality
-            conf.check_cc(header_name='stdint.h', define_name='OPJ_HAVE_STDINT_H',
-                          mandatory=False)
-
-            conf.define('OPJ_VERSION_MAJOR', '2', quote=False)
-            conf.define('OPJ_VERSION_MINOR', '3', quote=False)
-            conf.define('OPJ_VERSION_BUILD', '0', quote=False)
-
-        # check for the source tarball
-        if not exists(join(conf.path.abspath(), SOURCE + '.tar')):
-            conf.fatal('Missing OpenJPEG tarfile')
-
-        # untar and setup env
+    if j2kLayer == 'openjpeg':
         conf.env['MAKE_OPENJPEG'] = True
```

Comment:
Right now `MAKE_OPENJPEG` is always getting set... but shouldn't it only be getting set if you didn't provide `j2kHome`?  Otherwise `build()` is going to try to build it right?

Also, you can simplify and just set `MAKE_J2K` as a function of having j2kHome so it's in one spot...

Reply:
See modules/drivers/j2k/wscript:49

`MAKE_J2K` determines if we're building a J2K library at all. Then when we get into the specific wscript, `MAKE_OPENJPEG` and `MAKE_JASPER` are just verifying that we're building the one we're supposed to be building. Then `nitro/modules/c/j2k/wscript` uses `MAKE_OPENJPEG` and  `MAKE_JASPER` to check which library we have. 

If a prebuilt openjpeg is provided, we want MAKE_OPENJPEG set to say that we have openjpeg vs jasper, and MAKE_J2K=False (actually, just not set at all) to say that there's no need to actually build it.